feat: add article how-to/extract-packages.rst

dviererbe opened the pull request:
> [!IMPORTANT]
> This PR is targeted against #65 for better readability of the diff. **Set the base to the `2.0-preview` branch before merging!**

This adds a new how-to article that demonstrates how to extract the contained files in a source/binary package.

dviererbe committed:
feat: add article how-to/extract-packages.rst

Signed-off-by: Dominik Viererbe <dominik.viererbe@canonical.com>

rkratky commented on docs/how-to/extract-packages.rst:
```diff
@@ -0,0 +1,91 @@
+Extract packages
+================
+
+This article demonstrates how to extract the contents of debian packages.
```
```suggestion
This article demonstrates how to extract the contents of Debian packages.
```

rkratky commented on docs/how-to/extract-packages.rst:
```diff
@@ -0,0 +1,91 @@
+.. note::
+
+    A source package archive has the file ending `.dsc`.
+    See also the man page :manpage:`dsc(5)` for further information.
```
```suggestion
    See also the manual page :manpage:`dsc(5)` for further information.
```

rkratky commented on docs/how-to/extract-packages.rst:
```diff
@@ -0,0 +1,91 @@
+.. important::
+
+    Make sure that you have the `dpkg-dev` package installed. 
+    To install it run the following commands in a terminal:
```
```suggestion
    To install it, run the following commands in a terminal:
```

rkratky commented on docs/how-to/extract-packages.rst:
```diff
@@ -0,0 +1,91 @@
+
+        sudo apt update && sudo apt install dpkg-dev
+
+Run the follwong command in a terminal:
```
```suggestion
Run the following command in a terminal:
```

rkratky commented on docs/how-to/extract-packages.rst:
```diff
@@ -0,0 +1,91 @@
+    Make sure that you have the `dpkg-dev` package installed. 
+    To install it run the following commands in a terminal:
+
+    .. code-block:: bash
```
```suggestion
    .. code-block:: none
```

dviererbe replied:
Why did you change the syntax of this and other code blocks from `bash` to `none`?

rkratky replied:
Because:
- It's not semantically correct to use (Bash) syntax highlighting on command-line examples. It's not a Bash code block (i.e. a piece of Bash code where the user could expect to see syntax highlighting IRL).
- The highlighting that results from having `bash` specified is not particularly helpful. E.g.:
   
   ![image](https://github.com/user-attachments/assets/d5b64f7a-8ff5-42c6-90c3-e5d61ceed0d3)

   (It's not that bad in this example, but elsewhere, with numbers involved, for example, it becomes more confusing.)

But the main reason is that it's not intuitive for the user: they do not see their commands syntax-highlighted as they type/paste them on the command line.

rkratky commented on docs/how-to/extract-packages.rst:
```diff
@@ -0,0 +1,91 @@
+
+Run the follwong command in a terminal:
+
+.. code-block:: bash
```
```suggestion
.. code-block:: none
```

rkratky commented on docs/how-to/extract-packages.rst:
```diff
@@ -0,0 +1,91 @@
+    The path to the source package control file.
+
+``OUTPUT-DIRECTORY`` (optional)
+    The path to the directory where to extract the content of the source
+    package to. This directory **must not** exist. If no output directory is 
+    specified the content will be extracted into a directory named 
+    ``NAME-VERSION`` (where ``NAME`` is the name of the source package and 
+    ``VERSION`` its version) under the current working directory.
```
```suggestion
    The path to the directory where to extract the content of the source
    package to. This directory **must not** exist. If no output directory is 
    specified, the content is extracted into a directory named 
    ``NAME-VERSION`` (where ``NAME`` is the name of the source package and 
    ``VERSION`` its version) under the current working directory.
```

rkratky commented on docs/how-to/extract-packages.rst:
```diff
@@ -0,0 +1,91 @@
+    ``NAME-VERSION`` (where ``NAME`` is the name of the source package and 
+    ``VERSION`` its version) under the current working directory.
+
+See the man page :manpage:`dpkg-source(1)` for further informations.
```
```suggestion
See the manual page :manpage:`dpkg-source(1)` for further information.
```

rkratky commented on docs/how-to/extract-packages.rst:
```diff
@@ -0,0 +1,91 @@
+Extract a binary package
+------------------------
+
+This section demonstrates how to extract the content a binary packages.
```
```suggestion
This section demonstrates how to extract the content a binary package.
```

rkratky commented on docs/how-to/extract-packages.rst:
```diff
@@ -0,0 +1,91 @@
+
+.. note::
+
+    A binary package archive has the file ending `.deb`.
```
```suggestion
    A binary package archive has the file extension `.deb`.
```

rkratky commented on docs/how-to/extract-packages.rst:
```diff
@@ -0,0 +1,91 @@
+
+.. note::
+
+    A source package archive has the file ending `.dsc`.
```
```suggestion
    A source package archive has the file extension `.dsc`.
```

rkratky commented on docs/how-to/extract-packages.rst:
```diff
@@ -0,0 +1,91 @@
+.. note::
+
+    A binary package archive has the file ending `.deb`.
+    See also the man page :manpage:`deb(5)` for further information.
```
```suggestion
    See also the manual page :manpage:`deb(5)` for further information.
```

rkratky commented on docs/how-to/extract-packages.rst:
```diff
@@ -0,0 +1,91 @@
+    A binary package archive has the file ending `.deb`.
+    See also the man page :manpage:`deb(5)` for further information.
+
+Run the follwong command in a terminal:
```
```suggestion
Run the following command in a terminal:
```

rkratky commented on docs/how-to/extract-packages.rst:
```diff
@@ -0,0 +1,91 @@
+
+Run the follwong command in a terminal:
+
+.. code-block:: bash
```
```suggestion
.. code-block:: none
```

rkratky commented on docs/how-to/extract-packages.rst:
```diff
@@ -0,0 +1,91 @@
+
+``OUTPUT-DIRECTORY``
+    The path to the directory where to extract the content of the binary
+    package to. In comparison to :ref:`ExtractSourcePackage` this directory
```
```suggestion
    package to. In comparison to :ref:`ExtractSourcePackage`, this directory
```

rkratky commented on docs/how-to/extract-packages.rst:
```diff
@@ -0,0 +1,91 @@
+    package to. In comparison to :ref:`ExtractSourcePackage` this directory
+    can already exist and even contain files.
+
+See the man page :manpage:`dpkg-deb(1)` for further informations.
```
```suggestion
See the manual page :manpage:`dpkg-deb(1)` for further informations.
```

rkratky commented on docs/how-to/extract-packages.rst:
```diff
@@ -0,0 +1,91 @@
+
+.. tip::
+
+    Using ``--vextract`` instead of ``--extract`` will also display
```
```suggestion
    Using ``--vextract`` instead of ``--extract`` also outputs a list of
```

rkratky commented on docs/how-to/extract-packages.rst:
```diff
@@ -0,0 +1,91 @@
+    Using ``--vextract`` instead of ``--extract`` will also display
+    the extracted files to :term:`standard output <Standard Output>`.
+
+    To just list the contained files use the ``--contents`` option:
```
```suggestion
    To just list the files that the package contains, use the ``--contents`` option:
```

rkratky commented on docs/how-to/extract-packages.rst:
```diff
@@ -0,0 +1,91 @@
+
+    To just list the contained files use the ``--contents`` option:
+
+    .. code-block:: bash
```
```suggestion
    .. code-block:: none
```

rkratky commented on docs/how-to/extract-packages.rst:
```diff
@@ -0,0 +1,91 @@
+.. tip::
+
+    You can also replace ``dpkg-deb`` with ``dpkg`` for the examples 
+    demonstarted here. ``dpkg`` will forward the options to ``dpkg-deb``. 
+    See the man page :manpage:`dpkg(1)` for further informations.
```
```suggestion
    demonstrated here. ``dpkg`` forwards the options to ``dpkg-deb``. 
    See the manual page :manpage:`dpkg(1)` for further information.
```

rkratky reviewed (changes requested):
Thanks, @dviererbe. Some typo corrections and minor fixes suggested.

dviererbe committed:
fix(how-to/extract-packages): improve wording

Improves the wording of the how-to/extract-packages article and
corrects grammar/spelling mistakes.

Co-authored-by: Robert Krátký <rkratky@gmail.com>
Signed-off-by: Dominik Viererbe <dominik.viererbe@canonical.com>

dviererbe commented:
note: rebased onto `2.0-preview`

dviererbe committed:
fix: use correct file ext .deb for binary packages
